examples/basics/visuals/graph.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015, Vispy Development Team.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+
+"""
+This example demonstrates how to visualise a NetworkX graph using the
+GraphVisual.
+"""
+
+import sys
+
+import numpy as np
+import networkx as nx
+
+from vispy import app, gloo, visuals
+from vispy.visuals.graphs import layouts
+from vispy.visuals.transforms import STTransform
+
+
+class Canvas(app.Canvas):
+    def __init__(self):
+        app.Canvas.__init__(self, title="Simple NetworkX Graph",
+                            keys="interactive", size=(600, 600))
+
+        self.graph = nx.fast_gnp_random_graph(1000, 0.0006, directed=True)
+
+        self.visual = visuals.GraphVisual(
+            # np.asarray(nx.to_numpy_matrix(self.graph)),
+            nx.adjacency_matrix(self.graph),
```

Comment:
...and with the newer Python with your latest change, same thing but more informative error:
```
  File "/home/larsoner/custombuilds/vispy/vispy/visuals/graphs/layouts/force_directed.py", line 121, in _fruchterman_reingold
    delta_pos = self._calculate_delta_pos(adjacency_mat, pos, delta, t)
  File "/home/larsoner/custombuilds/vispy/vispy/visuals/graphs/layouts/force_directed.py", line 197, in _calculate_delta_pos
    )[:, :, np.newaxis]
  File "/home/larsoner/.local/lib/python2.7/site-packages/numpy/matrixlib/defmatrix.py", line 349, in __rmul__
    return N.dot(other, self)
ValueError: shapes (1000,1000,2) and (1000,1000,1) not aligned: 2 (dim 2) != 1000 (dim 1)
ERROR: Invoking <bound method Canvas.animate of <Canvas (PyQt4) at 0x7f682f8a3510>> for Event
```